app/routes.py

```diff
@@ -1,2 +1,77 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, abort, make_response, request, Response
 
+planets_bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+@planets_bp.route("", methods=["GET", "POST"])
+def handle_planets():
+    if request.method == "GET":
+
+        name_query = request.args.get("name")
+        if name_query:
+            planets = Planet.query.filter_by(name=name_query)           
+        else:
+            planets = Planet.query.all()
+
+        planets_response = []
+        for planet in planets:
+            planets_response.append({
+                "id": planet.id,
+                "name": planet.name,
+                "description": planet.description
+            })
+        return jsonify(planets_response)
+    
+    elif request.method == "POST":
```

Comment:
Older patterns for designing routes in Flask have shown a route that handles different REST methods and then uses if/elif to conditionally return the correct response. By separating the two request types out into their own methods, the code becomes more readable and each function has a single responsibility.